qiskit/circuit/quantumcircuit.py

```diff
@@ -176,6 +176,8 @@ def __init__(self, *regs, name=None, global_phase=0):
         self._global_phase = 0
         self.global_phase = global_phase
 
+        self.duration = None
```

Comment:
We also need methods on the circuit to get durations per qubit. See the `schedule.py` file which has `ch_start_time`, `ch_stop_time` and `ch_duration`. This is often more useful than the total circuit duration.

Reply:
Good suggestion. I like such functions. Can we leave it for future PR? To introduce such functions, we may need to consider subclassing (introducing a new class `ScheduledCircuit`) instead of using QuantumCircuit.duration as a flag.